### DISCUSSION TITLE
Type problem when using `plugins` option

### DISCUSSION THREAD

**Discussion by KiritaniAyaka** [vuepress/core#870] [Q&A] [answered]
I wrote the config file refer to [this document](https://v2.vuepress.vuejs.org/reference/plugin/register-components.html) but I'm using `vuepress.config.ts` instead of `vuepress.config.js`

My config file:
```ts
import { defineUserConfig } from 'vuepress'
import registerComponentsPlugin from '@vuepress/plugin-register-components'
import { path } from '@vuepress/utils'

const config = defineUserConfig({
	...
	plugins: [
		registerComponentsPlugin({
			componentsDir: path.resolve(__dirname, './components'),
		}),
	],
	...
})

module.exports = config
```

My vscode give me an error: `Type 'Plugin<PluginObject>' is not assignable to type 'PluginConfig<PluginOptions>'.`

I think this is because `plugins` option accept a type of `Plugin`, but `registerComponentsPlugin` function returned a type of `PluginConfig`

So does anyone know how do i fix it? Thanks in advance!

**Comment by meteorlxy** [comment, marked as answer]
I'm afraid that your `vuepress` version is lower than `2.0.0-beta.40`. Run `vuepress info` to check it.

**Reply by KiritaniAyaka** [reply]
Absolutely right, my `vuepress` version is `beta.39`. I solved it after upgrade vuepress. Thanks!
